nhost/service/manager.go

```diff
@@ -86,13 +95,162 @@ func (m *dockerComposeManager) SetGitBranch(gitBranch string) {
 	m.composeConfig = compose.NewConfig(m.nhostConfig, m.ports, m.env, gitBranch, m.projectName)
 }
 
+func (m *dockerComposeManager) checkPorts(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
+		return nil
+	default:
+	}
+
+	type composeSvc struct {
+		Service    string `json:"Service"`
+		Publishers []struct {
+			PublishedPort uint32 `json:"PublishedPort"`
+		}
+	}
+
+	cmd, err := compose.WrapperCmd(ctx, []string{"ps", "--filter", "status=running", "--format", "json"}, m.composeConfig, nil)
+	if err != nil && ctx.Err() != context.Canceled {
+		m.status.Error("Failed to list running services")
+		m.l.WithError(err).Debug("Failed to list running services")
+		return err
+	}
+
+	m.setProcessToStartInItsOwnProcessGroup(cmd)
+	out, err := cmd.CombinedOutput()
+	if err != nil && ctx.Err() != context.Canceled {
+		m.status.Error("Failed to list running services")
+		m.l.WithError(err).Debug("Failed to list running services")
+		return err
```

Comment:
That you always error with this tree lines is a bit of a code smell. For two reasons:

1. If you have specific steps to do every time you error you should consider a custom error that wraps around these steps
2. error message is the same in a few cases which will make it difficult to troubleshoot if something brakes